cloudwatch/cloudwatch.go

```diff
@@ -590,6 +624,40 @@ func (output *OutputPlugin) processRecord(e *Event) ([]byte, error) {
 	return data, nil
 }
 
+func (output *OutputPlugin) getECSMetadata() error {
+	ecsTaskMetadataEndpointV3 := os.Getenv("ECS_CONTAINER_METADATA_URI")
+	var metadata TaskMetadata
+	res, err := http.Get(fmt.Sprintf("%s/task", ecsTaskMetadataEndpointV3))
+	if err != nil {
+		logrus.Errorf("[cloudwatch %d] Failed to get the response from endpoint\n", output.PluginInstanceID)
+		return err
```

Comment:
Suggestion: Instead of logging an "error" here, wrap this error so the calling function can print it. Same for the next 3 lines. Something like this:
```suggestion
		return fmt.Errorf("failed to get endpoint response: %w", err)
```

This change requires an upgrade in go (which is in another PR somewhere); @PettitWesley should chime in if he likes this idea.

Reply:
I'm cool with this. We've had to delay our release till Monday because of upstream issues, so we have time to do the go version upgrade. 

I need to merge David's PR first, and then you can rebase and make this change.